### PR TITLE
Convert WLS images to the new matrix format

### PR DESCRIPTION
**Pull request by vovencij**
* Introduced notion of proprietary images to the build matrix. 
* Converted existing weblogic image creation scripts to the new matrix format. 
* In the webapp deployed to splunk distro test images, the span created from `@WithSpan` annotation is still present, because this test was legitimately broken on WLS 14/jdk11 until there was a fix for bytebuddy.

image

P.S. JDK in wls image names is weird because we use images build from oracle Dockerfiles and this is how it is there.

### CODE REVIEW

File: matrix/build.gradle.kts [hunk 1]
```diff
@@ -26,7 +26,11 @@ val dockerWorkingDir = project.buildDir.resolve("docker")
 
 val buildTestImagesTask = tasks.create("buildTestImages") {
```
Comment (iNikem) [inline matrix/build.gradle.kts:27]: As these are now built by Otel, can we remove this?

Comment (vovencij) [reply matrix/build.gradle.kts:27]: And leave only proprietary?

Comment (vovencij) [reply matrix/build.gradle.kts:27]: wait, liberty is also not present in otel?

Comment (vovencij) [reply matrix/build.gradle.kts]: Removed template and matrix definition, updated README.

**Review by iNikem — APPROVED** [review]

**Pull request merged**